airflow/providers/google/cloud/operators/workflows.py

```diff
@@ -0,0 +1,714 @@
+
+        # We are limited by allowed length of workflow_id so
+        # we use hash of whole information
+        exec_date = context['execution_date'].isoformat()
+        base = f"airflow_{self.dag_id}_{self.task_id}_{exec_date}_{hash_base}"
+        workflow_id = hashlib.md5(base.encode()).hexdigest()
+        return re.sub(r"[:\-+.]", "_", workflow_id)
+
+    def execute(self, context):
+        hook = WorkflowsHook(gcp_conn_id=self.gcp_conn_id, impersonation_chain=self.impersonation_chain)
+        workflow_id = self._workflow_id(context)
+
+        self.log.info("Creating workflow")
+        try:
+            operation = hook.create_workflow(
+                workflow=self.workflow,
+                workflow_id=workflow_id,
+                location=self.location,
+                project_id=self.project_id,
+                retry=self.retry,
+                timeout=self.timeout,
+                metadata=self.metadata,
+            )
+            workflow = operation.result()
+        except AlreadyExists:
+            workflow = hook.get_workflow(
+                workflow_id=workflow_id,
+                location=self.location,
+                project_id=self.project_id,
+                retry=self.retry,
+                timeout=self.timeout,
+                metadata=self.metadata,
+            )
+        return Workflow.to_dict(workflow)
+
+
+class WorkflowsUpdateWorkflowOperator(BaseOperator):
+    """
+    Updates an existing workflow.
+    Running this method has no impact on already running
+    executions of the workflow. A new revision of the
+    workflow may be created as a result of a successful
+    update operation. In that case, such revision will be
+    used in new workflow executions.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:WorkflowsUpdateWorkflowOperator`
+
+    :param workflow_id: Required. The ID of the workflow to be updated.
+    :type workflow_id: str
+    :param location: Required. The GCP region in which to handle the request.
+    :type location: str
+    :param project_id: Required. The ID of the Google Cloud project the cluster belongs to.
+    :type project_id: str
+    :param update_mask: List of fields to be updated. If not present,
+        the entire workflow will be updated.
+    :type update_mask: FieldMask
+    :param retry: A retry object used to retry requests. If ``None`` is specified, requests will not be
+        retried.
+    :type retry: google.api_core.retry.Retry
+    :param timeout: The amount of time, in seconds, to wait for the request to complete. Note that if
+        ``retry`` is specified, the timeout applies to each individual attempt.
+    :type timeout: float
+    :param metadata: Additional metadata that is provided to the method.
+    :type metadata: Sequence[Tuple[str, str]]
+    """
+
+    template_fields = ("workflow_id",)
+    template_fields_renderers = {"workflow": "json"}
```

Comment:
`workflow` is not listed in arguments. Shouldn't be supported alongside `update_mask`?